identity/backend/src/main/java/io/camunda/identity/usermanagement/service/UserService.java

```diff
@@ -40,64 +45,66 @@ public UserService(
     this.passwordEncoder = passwordEncoder;
   }
 
-  public CamundaUser createUser(final CamundaUserWithPassword userWithCredential) {
+  public CamundaUser createUserFailIfExists(
+      @Valid final CamundaUserWithPassword camundaUserWithPassword) {
     try {
-      final UserDetails userDetails =
-          User.withUsername(userWithCredential.getUsername())
-              .password(userWithCredential.getPassword())
-              .passwordEncoder(passwordEncoder::encode)
-              .disabled(!userWithCredential.isEnabled())
-              .roles("DEFAULT_USER")
-              .build();
-      camundaUserDetailsManager.createUser(userDetails);
-      final var createdUser = userProfileRepository.findByUsername(userDetails.getUsername());
-      userProfileRepository.save(
-          new Profile(
-              createdUser.getId(), userWithCredential.getEmail(), userWithCredential.getName()));
-      return userProfileRepository.findByUsername(userWithCredential.getUsername());
+      return createUser(camundaUserWithPassword);
     } catch (final DuplicateKeyException e) {
-      throw new RuntimeException("user.duplicate");
+      throw new IllegalArgumentException("User invalid.");
     }
   }
 
-  public void deleteUser(final Long id) {
+  public CamundaUser createUserIfNotExists(
+      @Valid final CamundaUserWithPassword camundaUserWithPassword) {
+    if (camundaUserDetailsManager.userExists(camundaUserWithPassword.getUsername())) {
+      return findUserByUsername(camundaUserWithPassword.getUsername());
+    }
+
+    return createUser(camundaUserWithPassword);
+  }
+
+  public void deleteUser(@NotNull(message = "User invalid.") final Long id) {
```

Comment:
💭 This may be because its handled somewhere else that I'm not aware of but I'd be a bit more descriptive here
```suggestion
  public void deleteUser(@NotNull(message = "Invalid user ID.") final Long id) {
```

Reply:
Agreed ... this is a clear case, cannot hide what was invalid :-D ... changed :)